app/src/main/java/de/westnordost/streetcomplete/map/components/TracksMapComponent.kt

```diff
@@ -21,47 +21,44 @@ class TracksMapComponent(ctrl: KtMapController) {
     private val layer2 = ctrl.addDataLayer(LAYER2)
 
     private var index = 0
-    private var tracks: MutableList<MutableList<LngLat>>
-
-    init {
-        tracks = ArrayList()
-        tracks.add(ArrayList())
-    }
+    private data class Track(val trackpoints: MutableList<LngLat>, val isRecording: Boolean)
+    private var tracks: MutableList<Track> = arrayListOf(Track(ArrayList(), false))
 
     /** Add a point to the current track */
     fun addToCurrentTrack(pos: Location) {
         val track = tracks.last()
-        track.add(pos.toLngLat())
+        track.trackpoints.add(pos.toLngLat())
+        val trackpoints = track.trackpoints
 
         // every 100th trackpoint, move the index to the back
-        if (track.size - index > 100) {
+        if (trackpoints.size - index > 100) {
             putAllTracksInOldLayer()
         } else {
-            layer1.setFeatures(listOf(track.subList(index, track.size).toPolyline(false)))
+            layer1.setFeatures(listOf(trackpoints.subList(index, trackpoints.size).toPolyline(false, track.isRecording)))
         }
     }
 
     /** Start a new track. I.e. the points in that track will be drawn as an own polyline */
-    fun startNewTrack() {
-        tracks.add(ArrayList())
+    fun startNewTrack(record: Boolean) {
+        tracks.add(Track(ArrayList(), record))
         putAllTracksInOldLayer()
     }
 
     /** Set all the tracks (when re-initializing) */
     fun setTracks(tracks: List<List<Location>>) {
-        this.tracks = tracks.map { track -> track.map { it.toLngLat() }.toMutableList() }.toMutableList()
+        this.tracks = tracks.map { track -> Track(track.map { it.toLngLat() }.toMutableList(), false) }.toMutableList()
```

Comment:
Yes, as to my understanding this reloading is only when the app is re-opened. In this case, no recording should be active, and I am not sure how useful old recording segments will be compared to just showing were you have been (basically I didn't feel it was worth adding this).

Reply:
It's when the view is recreated. This can happen any time when a user tabs out and then in again.

Reply:
Oh, let me test and get back to you on this then.

Reply:
**can** - it will be hard to reproduce because the Android system decides when to do that and when to not do it (based on memory constraints or whatever). I know one way to "force" this - remove from the AndroidManifest that the MainActivity handles `orientation` changes itself for testing. If it does not handle it itself, rotating the phone will recreate the view.

Reply:
It looks like this also stops the recording since this isn't something that is stored in memory. It took a few turns, but eventually it recreated the view and you are correct that it looses the track history of recordings, but it also stops the recording. I am not sure how best to fix, as I think I need to record someplace that the user has started recording a track? or maybe it isn't worth fixing this? thoughts?

https://user-images.githubusercontent.com/2222562/149715792-f86758ac-65b7-4539-95e4-46bc25df4542.mp4

Reply:
Hmm,

for the use case that one just wants to add a GPX track to a note showing a short driveway/connecting path, I think it is okay if the recording is cancelled when the user changes apps and changes back after some time because the record feature can be understood to be "in the process of creating a note". Tabbing out in-between can be interpreted as that the user doesn't want to create a note after all.

On the other hand, users may want to add longer records as well, to create notes like "I walked every path in this forest/park, some paths seem to be missing!" or whatever. So in this case, the record would go on for a while and it is normal during a mapping session to sometimes tab out to some other app. Every tabbing out would then endanger that the whole track is being discarded (depending on how aggressive the Android system will destroy apps not in the foreground).

So, in conclusion, I would say that the recording track should indeed be saved into the instance state.

Reply:
I think the user would still expect that the track keeps recording if they switch apps for a few seconds. How do you recommend I record the state in the fragment? I am not sure how to even approach in implementation.

*Side comment: There is the additional case that switching apps will stop getting location information since the app is in the background now and we only have location info when in the forground.*

Reply:
> here is the additional case that switching apps will stop getting location information since the app is in the background now and we only have location info when in the forground.

Yes I know, this is deliberate though.